nixos/modules/services/networking/fedimintd.nix

```diff
@@ -0,0 +1,357 @@
+          default = "signet";
+          example = "bitcoin";
+          description = "Bitcoin network to participate in.";
+        };
+        rpc = {
+          address = mkOption {
+            type = types.str;
+            default = "http://127.0.0.1:38332";
+            example = "signet";
+            description = "Bitcoin node (bitcoind/electrum/esplora) address to connect to";
+          };
+
+          kind = mkOption {
+            type = types.str;
+            default = "bitcoind";
+            example = "electrum";
+            description = "Kind of a bitcoin node.";
+          };
+
+          secretFile = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = ''
+              If set the URL specified in `bitcoin.rpc.address` will get the content of this file added
+              as an URL password, so `http://user@example.com` will turn into `http://user:SOMESECRET@example.com`.
+
+              Example:
+
+              `/etc/nix-bitcoin-secrets/bitcoin-rpcpassword-public` (for nix-bitcoin default)
+            '';
+          };
+        };
+      };
+
+      consensus.finalityDelay = mkOption {
+        type = types.number;
+        default = 10;
+        description = "Consensus peg-in finality delay.";
+      };
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/fedimintd-${name}/";
+        readOnly = true;
+        description = ''
+          Path to the data dir fedimintd will use to store its data.
+          Note that due to using the DynamicUser feature of systemd, this value should not be changed
+          and is set to be read only.
+        '';
+      };
+
+      nginx = {
+        enable = mkEnableOption "fedimint";
+        config = mkOption {
+          type = types.nullOr (types.submodule (
+            recursiveUpdate (import ../web-servers/nginx/vhost-options.nix { inherit config lib; }) {
+              options.serverName = {
+                default = "radicle-${config.networking.hostName}.${config.networking.domain}";
+                defaultText = "radicle-\${config.networking.hostName}.\${config.networking.domain}";
+              };
+            }
+          ));
+          default = null;
+          description = "Overrides to the nginx vhost section for api";
+        };
+      };
+    };
+  };
+in
+{
+  options = {
+    services.fedimintd = mkOption {
+      type = types.attrsOf (types.submodule fedimintdOpts);
+      default = { };
+      description = "Specification of one or more fedimintd instances.";
+    };
+  };
+
+  config = mkIf (eachFedimintd != { }) {
+
+    assertions = flatten
+      (mapAttrsToList
+        (fedimintdName: cfg: [
+          {
+            assertion = cfg.p2p.address != null;
+            message = ''
+              `services.fedimintd.${fedimintdName}.p2p.address` must be set to address reachable by other peers.
+
+              Example: `fedimint://p2p.mymint.org`.
```

Comment:
Address could here also refer to an ip address. Maybe we should use URI?

Reply:
Good point. I went with `url`, as we do require `protocol://name`